openpype/hosts/houdini/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from openpype.pipeline import PublishValidationError
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.houdini.api.action import SelectInvalidAction
+
+import hou
+
+
+class HotFixAction(RepairAction):
+    """Set End frame to the minimum valid value."""
+
+    label = "End frame hotfix"
+
+
+class ValidateFrameRange(pyblish.api.InstancePlugin):
+    """Validate Frame Range.
+
+    Due to the usage of start and end handles,
+    then Frame Range must be >= (start handle + end handle)
+    which results that frameEnd be smaller than frameStart
+    """
+
+    order = pyblish.api.ValidatorOrder - 0.1
+    hosts = ["houdini"]
+    label = "Validate Frame Range"
+    actions = [HotFixAction, SelectInvalidAction]
+
+    def process(self, instance):
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            nodes = [n.path() for n in invalid]
+            raise PublishValidationError(
+                "Invalid Frame Range on: {0}".format(nodes),
+                title="Invalid Frame Range"
+            )
+
+    @classmethod
+    def get_invalid(cls, instance):
+
+        if not instance.data.get("instance_node"):
+            return
+
+        rop_node = hou.node(instance.data["instance_node"])
+        if instance.data["frameStart"] > instance.data["frameEnd"]:
+            cls.log.error(
+                "Wrong frame range, please consider handle start and end.\n"
+                "frameEnd should at least be {}.\n"
+                "Use \"End frame hotfix\" action to do that."
+                .format(
+                    instance.data["handleEnd"] +
+                    instance.data["handleStart"] +
+                    instance.data["frameStartHandle"]
+                )
```

Comment:
Would it make sense to be way more descriptive?

Questions I have when reading this:

- What are the expected handles of the asset? (Like what is it applying or validating against)
- It should likely describe in the `description` of the `PublishValidationError` that the handles are **inclusive**. The render rop node's frame range should _include_ the handles - they are not added before start and after end frames. That description should then **also** state that it's also possible to disable using/applying handles of the asset document.

I still feel we're better off adding `handleStart` and `handleEnd` attributes on the instances so I just wanted to mention that again - but feel free to ignore it if you feel that has issues of its own.

Reply:
> if you feel that has issues of its own.

it's more about avoiding possible issues and considering possible future updates ideas. 

I really don't know if each (render)instance should include its own `handleStart` and `handleEnd` of course we can try adding them and test in production what will happen. 
do think it may cause a mess as artists my write values that make sense to them ?